homeassistant/components/mqtt/client.py

```diff
@@ -494,6 +537,18 @@ def _async_untrack_subscription(self, subscription: Subscription) -> None:
         except (KeyError, ValueError) as ex:
             raise HomeAssistantError("Can't remove subscription twice") from ex
 
+    async def _async_queue_subscriptions(
+        self, subscriptions: Iterable[tuple[str, int]]
+    ) -> None:
+        """Queue requested subscriptions."""
+        async with self._paho_lock:
```

Comment:
Why do we need to take the paho lock here, even though we don't talk to the paho client?

Reply:
Because we change the pending subscriptions here. If any subscribe process is still running we will wait until that is finished and has cleared the dict.

Reply:
I could use an other that the paho lock for guarding the pending subscriptions.

Reply:
Yes, please use a lock which specifically protects `self._pending_subscriptions` instead👍